### PR TITLE
Assorted Caffe2 importer support for seq2seq

### PR DESCRIPTION
**Pull request by jfix71**
Each commit is pretty self explanatory.

### CODE REVIEW

File: lib/Graph/Graph.cpp [hunk 1]
```diff
@@ -1299,6 +1299,14 @@ ScatterAssignNode *Function::createScatterAssign(llvm::StringRef name,
   return addNode(new ScatterAssignNode(name, data, indices, slices));
 }
 
+Variable *Function::createShape(llvm::StringRef name, NodeValue input) {
```
Comment (artemrakhov) [inline lib/Graph/Graph.cpp]: I am not convinced that this should be in Graph API. Would you use it if you were to create graph using C++ API?

File: lib/Importer/Caffe2.cpp [hunk 2]
```diff
@@ -415,6 +447,65 @@ void caffe2ModelLoader::loadOperator(const caffe2::OperatorDef &op) {
     return;
   }
 
+  if (typeName == "Tanh") {
```
Comment (artemrakhov) [inline lib/Importer/Caffe2.cpp]: Can we have this and other ops in CommonOperatorLoader? I think they have the same semantics in ONNX.

File: lib/Importer/Caffe2.cpp [hunk 3]
```diff
@@ -470,12 +561,23 @@ void caffe2ModelLoader::loadWeights(caffe2::NetDef &net) {
       if (dict["values"]->floats_size()) {
         assert(typeName != "GivenTensorIntFill" &&
                typeName != "GivenTensorInt64Fill");
+
+        // Assume a 1D vector of length of input values.
+        if (dim.size() == 0) {
+          dim.push_back(dict["values"]->floats_size());
```
Comment (artemrakhov) [inline lib/Importer/Caffe2.cpp]: This is 0-dims Tensor, right? Why can't we leave it as 0 dims? Have you seen cases where `dict["values"]->floats_size()` is more than 1?

File: lib/Importer/Caffe2.cpp (hunk 2 above)
Comment (jfix71) [reply lib/Importer/Caffe2.cpp]: I can move it/others, but I have no ONNX proto with these operators to make sure they work correctly, so I was conservatively adding only for C2.

**Comment by jfix71** [comment]
I'm going to just directly create and set the values for the Shape loader. I think `createVariableFromValue()` is a good idea but it would likely be used very widely and so it's not in scope for this PR.

File: lib/Importer/Caffe2.cpp (hunk 2 above)
Comment (jfix71) [reply lib/Importer/Caffe2.cpp]: I moved all but Slice -- there is a slight difference in the ONNX/C2 implementations for the start and end indices and negative values, so I'd rather not add it in without having a proto with it to test that it works.

**Pull request merged**